Automattic/jetpack
# Widget Visibility: Fix "Match all" setting being overwritten on saving in the block editor.

Fixes #22249

#### Changes proposed in this Pull Request:

Checking for a non-empty value is necessary when saving in the widget block editor. I don't know exactly what changed in the block widgets update process, but this solves the issue at hand. 

#### Jetpack product discussion


No

#### Does this pull request change what data or activity we track or use?


No 

#### Testing instructions:





This is how to reproduce the issue and verify the fix: 

- Install Jetpack
- Instal Classic Widgets plugin
- Enable Jetpack's Widget Visibility
- Add Categories widget to a widget area and add two conditions (I did show on page and post)
- Make sure visibility is not set to "match all conditions"
- Disable Classic Widgets plugin
- Go back to the Widgets and open the "Categories" widget
- Change something (such as dropdown) and update
- Navigate away and then back to the widgets screen and "Match All Conditions" will be checked
